classes/class-p4-settings.php

```diff
@@ -0,0 +1,161 @@
+<?php
+
+if ( ! class_exists( 'P4_Settings' ) ) {
+	/**
+	 * Class P4_Settings
+	 */
+	class P4_Settings {
+
+		/**
+		 * Option key, and option page slug
+		 * @var string
+		 */
+		private $key = 'planet4_options';
+
+		/**
+		 * Array of metaboxes/fields
+		 * @var array
+		 */
+		protected $option_metabox = array();
+
+		/**
+		 * Options Page title
+		 * @var string
+		 */
+		protected $title = '';
+
+		/**
+		 * Options Page hook
+		 * @var string
+		 */
+		protected $options_page = '';
+
+		/**
+		 * Constructor
+		 */
+		public function __construct() {
+
+			// Set our title
+			$this->title   = __( 'Planet4', 'planet4-master-theme' );
+			$pages         = get_pages();
+
+			$pages_array[] = __( 'Select Page' ,'planet4-master-theme' ) ;
+			foreach ( $pages as $all_pages ) {
+				$pages_array[$all_pages->ID] = __( $all_pages->post_title ,'planet4-master-theme' ) ;
+			}
+
+			$categories         = get_categories();
+
+			$categories_array[] = __( 'Select Category' ,'planet4-master-theme' ) ;
+			foreach ( $categories as $category ) {
+				$categories_array[$category->term_id] = __( $category->cat_name ,'planet4-master-theme' );
+			}
```

Comment:
@sagarsdeshmukh Don't we want to be able to choose the 'parent' category, so that we
can then get all the children categories dynamically like referenced [here](https://github.com/greenpeace/planet4-plugin-blocks/blob/5c8f29efe07e6bf490896013ab06006cc5bb37ee/classes/controller/blocks/class-p4bks-blocks-splittwocolumns-controller.php#L24)

Reply:
@angtheod Yes, Angelos if here we don't choose parent category then it will work as per reference. [here](https://github.com/greenpeace/planet4-plugin-blocks/blob/feature/1411-make-dynamic-p4-settings/classes/controller/blocks/class-p4bks-blocks-splittwocolumns-controller.php#L24)

Reply:
@sagarsdeshmukh @koyan 
Hey Sagar, I think we need to give them the option to select the parent category (for example called 'Issues') and not the children categories.
Also, I think it would be better to name the drop down label 'Select Issues parent category' instead of just 'Select category'.

Reply:
@sagarsdeshmukh @koyan 
Hey Sagar, I think you are getting only the categories that have at least 1 post assigned to them or that have children categories. I believe we want all categories to be listed in there.

It is worth noting that we don't need to use CMB2 for these fields. WP has 2 built-in functions that do exactly what we want (and better actually, since they listed options inside the drop down can be displayed in hierarchical way).

Take a look at those 2 functions:

`wp_dropdown_pages( [
				'hide_empty'   => 0,
				'hierarchical' => true,
			] );
			wp_dropdown_categories( [
				'hide_empty'   => 0,
				'hierarchical' => true,
				'orderby'      => 'name',
			] );`

Also, I think it would be better to name the drop down label 'Select Issues parent category' instead of just 'Select category'.